thunder/core/transforms.py

```diff
@@ -3156,8 +3159,12 @@ def recompute_saved_for_backward(fwd_trace: Trace, bwd_trace: Trace) -> tuple[Tr
             if prod_arg in fwd_trace_args:
                 required_fw_args.add(prod_arg)
         for prod_out in prod.flat_outs:
+            if isinstance(prod_out, TensorProxy):
+                swap_map[variableify(prod_out)] = prod_out.replace_name(f"remat_for_{prod_out.name}")
```

Comment:
What does "remat_for" indicate in a full sentence? I think using "of" is a more suitable preposition here.

Could you please update the `replace_name` method to add this name to the current trace:
https://github.com/Lightning-AI/lightning-thunder/blob/1de5dd81d80a6c6036eeb9f77aa1116cb429fbc3/thunder/core/proxies.py#L178

Reply:
What do you mean by that? Are you suggesting to change the `replace_name` method by adding a default value "remat_of_" ?